### PR TITLE
Filters by active in content items data source

### PR DESCRIPTION
**Pull request by vinnyjth**
### What does this PR do, or why is it needed?
In the past, items would show up in the app before they went live or after they expire. Many churches use those dates to force content to go live or to remove content from their feed, so we need to respect the values of those properties. This PR does that. 

I also added some methods to the `request-builder.js` so that request filter chaining should be more concise and read better.

### What design trade-offs/decisions were made?
n/a

### How do I test this PR?
1. Open the app w/ local API.
2. Examine the content item feed and pick an item.
3. In Rock, edit the Active At or Expires At dates to make the item inactive.
4. Refresh the content feed in the app, the item should disappear.  

### What automated tests did you write?

Updated existing tests.

## TODO:

- [ ] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [tag-issues-here]
- [ ] No new warnings in tests, in storybook, and in-app
- [ ] Upload GIF(s) of iOS and Android
- [ ] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by redreceipt** [comment]
@vinnyjth what's the instance of Rock this is pulling from? apollosrock.newspring.cc doesn't seem like the right content?
screen shot 2018-12-13 at 8 46 00 pm

**Comment by vinnyjth** [comment]
@redreceipt the different content is due to all of the content you normally see being expired :P w/ we need to login to the rock instance and change the dates on all content that _should_ show up in the app now.

**Comment by conrad-vanl** [comment]
@vinnyjth I'd say that content _really_ looks like the content from NewSpring's rock instance, not Apollo's?

**Comment by vinnyjth** [comment]
@conrad-vanl True. @redreceipt, I'm assuming you have the wrong values in your .env file. Compare what you have against one of your teammates values, or the values in basecamp.

File: packages/apollos-data-connector-rock/src/content-items/data-source.js [hunk 1]
```diff
@@ -2,6 +2,8 @@ import RockApolloDataSource from '@apollosproject/rock-apollo-data-source';
 import ApollosConfig from '@apollosproject/config';
 
 const { ROCK_MAPPINGS } = ApollosConfig;
+const LIVE_CONTENT = (date = new Date()) =>
```
Comment (conrad-vanl) [inline packages/apollos-data-connector-rock/src/content-items/data-source.js]: I'm fairly certain that `new Date()` gets evaluated once at initial app boot, so once the date changes this default value will be wrong...I think you _have_ to do something like:

```
const LIVE_CONTENT = (date) => {
   if (!date) date = new Date();
```

Comment (vinnyjth) [reply packages/apollos-data-connector-rock/src/content-items/data-source.js]: @conrad-vanl ooooh good catch, but I think in Javascript we are okay
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters#Evaluated_at_call_time

**Comment by conrad-vanl** [comment]
Actually, it might appear that I'm incorrect, as I think this is doing the same thing: 

image

File: packages/apollos-data-connector-rock/src/content-items/data-source.js (hunk 1 above)
Comment (conrad-vanl) [reply packages/apollos-data-connector-rock/src/content-items/data-source.js]: yeah! I'm noticing that now, okay, back to 👀  i'll have this approved shortly me thinks

**Comment by vinnyjth** [comment]
@conrad-vanl yep - MDN says that args defaults are evaluated at runtime
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters#Evaluated_at_call_time

Which honestly I didn't even consider, so I"m glad you brought it up.

**Pull request merged**